### PR TITLE
SRV-117 - Profile Token Renew

### PR DESCRIPTION
**Pull request by bgelatti**
## What does PR do?

Created a route to renew token without the necessity of login again for max 24 hours

## Type of alteration

- [ ] Bug fix
- [X] New feature
- [ ] Breaking change
- [ ] Documentation update

### CODE REVIEW

File: src/modules/Resources/Profile.ts [hunk 1]
```diff
@@ -303,6 +303,28 @@ class Profile extends TagoIOModule<GenericModuleParams> {
...
+   */
+  public async renewProfileToken(profileID: GenericID, token: GenericToken): Promise<TokenCreateResponse> {
+    let result = await this.doRequest<TokenCreateResponse>({
+      path: `/profile/${profileID}/token/renew`,
+      method: "POST",
+      body: {
+        name: "Generated by TagoIO Renew",
+        profile_id: profileID,
+        origin: "admin",
+        old_token: token,
```
Comment (klausborges) [inline src/modules/Resources/Profile.ts:319]: Small things to consider in the request `body`:

-  `profile_id` property is already in route params
- `old_token` is already in the headers

Usage:

```ts
// current version
const resource = new Resources({ token: "my-token" });
await resource.profiles.renewProfileToken(profileID, "my-token");

// proposed changes
const resource = new Resources({ token: "my-token" });
await resource.profiles.renewProfileToken(profileID);
```

Comment (klausborges) [reply src/modules/Resources/Profile.ts:319]: @bgelatti if this was already discussed, just let me know!

**Review by otaaviostein — APPROVED** [review]

File: src/modules/Resources/Profile.ts (hunk 1 above)
Comment (bgelatti) [reply src/modules/Resources/Profile.ts:319]: the code of the backend was implemented in that way, to change this, need to change de back too and probably the front

Comment (bgelatti) [reply src/modules/Resources/Profile.ts]: done

**Pull request closed without merging**